src/appleseed/renderer/kernel/lighting/lighttypes.h

```diff
@@ -95,55 +99,70 @@ class LightSource
     // Destructor.
     virtual ~LightSource() {}
 
+    enum SourceTypes
+    {
+        NPL = 0,
```

Comment:
We use NPL and EMT among ourselves but they aren't super clear terms. I'd suggest longer names: `NonPhysicalLight` and `EmittingTriangle`.